src/test/ui/rfc-3086-metavar-expr/syntax-errors.stderr

```diff
@@ -0,0 +1,129 @@
+note: try removing `$`
+  --> $DIR/syntax-errors.rs:17:39
+   |
+LL |     ( $( $i:ident ),* ) => { ${ count($i) } };
+   |                                       ^
+
+error: could not find an expected `ident` element. Probably some number or special character was declared so try removing any arbitrary prefixes
+  --> $DIR/syntax-errors.rs:17:33
+   |
+LL |     ( $( $i:ident ),* ) => { ${ count($i) } };
+   |                                 ^^^^^
+
+error: expected one of: `*`, `+`, or `?`
+  --> $DIR/syntax-errors.rs:17:31
+   |
+LL |     ( $( $i:ident ),* ) => { ${ count($i) } };
+   |                               ^^^^^^^^^^^^^
+
```

Comment:
I appreciate the time you've put into improving the errors, but I feel like these error messages still need work. @estebank or @petrochenkov are better suited to give you pointers on how to do it right.

1. The note should be attached to a particular error. I believe your use of `span_note_without_error` is the problem here.
2. I don't find the second error very helpful. If I received that error as a user, I wouldn't really know what to do. Also, we typically don't put suggestions in the error message itself.
3. ```expected one of: `*`, `+`, or `?` ``` This seems wrong. I feel like the error message I would expect is something like:

```
error: expected ident found `$`
  --> $DIR/syntax-errors.rs:17:31
   |
LL |     ( $( $i:ident ),* ) => { ${ count($i) } };
   |                                       ^
```

Reply:
> expected one of: `*`, `+`, or `?` This seems wrong. I feel like the error message I would expect is something like:

`parse_tree` now returns early to avoid unrelated meta-variables expression errors -> https://github.com/rust-lang/rust/pull/93545/files#diff-152c18de02d7a7e6ad577226b23ae10329c206af65287b78f35cc7f37e7ba699R187-R189

> The note should be attached to a particular error. I believe your use of span_note_without_error is the problem here.

> I don't find the second error very helpful. If I received that error as a user, I wouldn't really know what to do. Also, we typically don't put suggestions in the error message itself.

![Capture d’écran de 2022-02-22 18-40-00](https://user-images.githubusercontent.com/17877264/155224322-0a819da2-2583-47ec-8d63-db2179038ed6.png)

Reply:
I'm fine with the errors as they are.